packages/esm-patient-chart-app/src/clinical-views/encounter-tile/encounter-tile.component.tsx

```diff
@@ -1,84 +1,79 @@
-              encounterType={encounterTypeUuid}
-              columns={columns}
-            />
-          ))}
-        </Column>
+        <Grid fullWidth>
+          {columns.map((column, index) => {
+            return (
+              <Column
+                key={`${column.encounterTypeUuid}-${column.title}-${index}`}
+                sm={columnSpan}
+                md={columnSpan}
+                lg={columnSpan}
+              >
+                <EncounterData patientUuid={patientUuid} column={column} />
+              </Column>
+            );
+          })}
+        </Grid>
       </Tile>
     </Layer>
   );
 });
 
 const EncounterData: React.FC<{
   patientUuid: string;
-  encounterType: string;
-  columns: EncounterTileColumn[];
-}> = ({ patientUuid, encounterType, columns }) => {
-  const { lastEncounter, isLoading, error, isValidating } = useLastEncounter(patientUuid, encounterType);
+  column: EncounterTileColumn;
+}> = ({ patientUuid, column }) => {
   const { t } = useTranslation();
+  const { lastEncounter, isLoading, error, isValidating } = useLastEncounter(patientUuid, column.encounterTypeUuid);
+  const obsValue = column.getObsValue(lastEncounter);
+  const summaryValue =
+    column.hasSummary === true && column.getSummaryObsValue && typeof column.getSummaryObsValue === 'function'
+      ? column.getSummaryObsValue(lastEncounter)
+      : null;
 
   if (isLoading || isValidating) {
     return <CodeSnippetSkeleton type="multi" className="skeleton" />;
   }
 
   if (error || lastEncounter === undefined) {
     return (
-      <div className={styles.tileBox}>
-        {columns.map((column, ind) => (
-          <div key={ind}>
-            <span className={styles.tileTitle}>{t(column.title)}</span>
-            <span className={styles.tileValue}>{error?.message}</span>
-          </div>
-        ))}
-      </div>
+      <>
+        <span className={styles.tileTitle}>{t(column.title)}</span>
+        <span className={styles.tileValue}>{error?.message}</span>
+      </>
     );
   }
 
   return (
-    <div className={styles.tileBox}>
-      {columns.map((column, ind) => {
-        return (
-          <div key={ind}>
-            <span className={styles.tileTitle}>{t(column.header)}</span>
-            <span className={styles.tileValue}>
-              <p>{column.getObsValue(lastEncounter)}</p>
-            </span>
-            {column.hasSummary && (
-              <>
-                <span className={styles.tileValue}>
-                  <p>{column.getSummaryObsValue(lastEncounter)}</p>
-                </span>
-                <span className={styles.tileValue}>
-                  <p>{column.getObsValue(lastEncounter)}</p>
-                </span>
-              </>
-            )}
-          </div>
-        );
-      })}
-    </div>
+    <>
+      <span className={styles.tileTitle}>{t(column.header)}</span>
+      {!(obsValue === '--' && summaryValue !== '--' && summaryValue !== null) && (
```

Comment:
Maybe?

```suggestion
      {!(obsValue === '--' && summaryValue !== '--' && isNil(summaryValue)) && (
```

`isNil(x)` is basically `typeof x === 'undefined' || x === null`